src/processors/pinContentToIpfs.ts

```diff
@@ -0,0 +1,80 @@
+import * as IPFS from "ipfs-http-client";
+import createProcessor from "../processor";
+
+export type FileContent = Uint8Array;
+
+export class Pinner {
+  ipfsOptions: IPFS.Options;
+  ipfs: IPFS.IPFSHTTPClient;
+  where: string;
+
+  constructor(ipfsOptions = {}, where = "local") {
+    this.ipfsOptions = ipfsOptions;
+    this.ipfs = IPFS.create(this.ipfsOptions);
+    this.where = where;
+  }
+
+  async pin(content: FileContent) {
+    if (typeof content === "string") return content;
+
+    const pinningService = await this.#_getRemotePinningService();
+    const path = "";
+
+    const { cid } = await this.ipfs.add(
+      { content, path },
+      { wrapWithDirectory: false }
+    );
+
+    if (this.where === "local") {
+      await this.ipfs.pin.add(cid);
+    } else {
+      try {
+        await this.ipfs.pin.remote.add(cid, {
+          service: pinningService.service,
+          name: "Tableland Upload",
+        });
+      } catch (err: any) {
+        const message = err.message;
+        if (!message.includes("DUPLICATE_OBJECT")) {
+          throw err;
+        }
+
+        console.log(
+          "Good news; that CID is already pinned to your pinning service."
+        );
+      }
+    }
+
+    return cid.toV1().toString();
+  }
+
+  async resolveCid(cid: string) {
+    return await this.ipfs.get(cid);
+  }
+
+  async #_getRemotePinningService() {
+    const pinningServices = await this.ipfs.pin.remote.service.ls();
+
+    if (pinningServices.length < 1) {
+      throw new Error("No remote pinning service connected.");
+    }
+
+    // TODO: why are we using the first value?
+    return pinningServices[0];
```

Comment:
Added a todo comment here because I'm not really sure what the implications of 2 remote services existing are.  Regardless, this preserves the existing logic.

Reply:
I wasn't sure either. Perhaps we should loop through and pin to all, since a huge purpose for IPFS is to decentralize. Or pin to all by default and make any option? Don't need to change now, but worth thinking about.